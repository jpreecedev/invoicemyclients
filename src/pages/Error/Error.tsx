import React from 'react'

const ErrorPage: React.FC = () => {
  return (
    <section>
      <div className="container d-flex flex-column">
        <div className="row align-items-center justify-content-center no-gutters min-vh-100">
          <div className="col-12 col-md-6 py-8 py-md-11">
            <h1 className="display-3 font-weight-bold text-center">Error.</h1>

            <p className="mb-5 text-center text-muted">
              We ran into an issue, for which we apologise. We have logged the error and we will
              investigate soon.
            </p>

            <div className="text-center">
              <a className="btn btn-primary" href="#!">
                Back to safety
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
  )
}

export { ErrorPage }
