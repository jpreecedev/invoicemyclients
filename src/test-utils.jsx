import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
