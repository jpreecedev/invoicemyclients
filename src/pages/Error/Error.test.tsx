import React from 'react'
import { renderWithRouter } from '../../test-utils'

import { ErrorPage } from './Error'

describe('<Error /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<ErrorPage />).container).toMatchSnapshot()
  })
})
