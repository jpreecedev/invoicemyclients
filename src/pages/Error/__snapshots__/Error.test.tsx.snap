// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> tests should match snapshot 1`] = `
<div>
  <section>
    <div
      class="container d-flex flex-column"
    >
      <div
        class="row align-items-center justify-content-center no-gutters min-vh-100"
      >
        <div
          class="col-12 col-md-6 py-8 py-md-11"
        >
          <h1
            class="display-3 font-weight-bold text-center"
          >
            Error.
          </h1>
          <p
            class="mb-5 text-center text-muted"
          >
            We ran into an issue, for which we apologise. We have logged the error and we will investigate soon.
          </p>
          <div
            class="text-center"
          >
            <a
              class="btn btn-primary"
              href="#!"
            >
              Back to safety
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
